### PR TITLE
(PDB-4872) Update harmful language

### PR DESCRIPTION
**Pull request by Zak-Kent**
This still needs doc updates but should be ready for another set of eyes. I can squash the first couple commits into one if we prefer.

### CODE REVIEW

File: src/puppetlabs/puppetdb/middleware.clj [hunk 1]
```diff
@@ -41,25 +41,25 @@
...
+  certificate allowlist (see `cn-whitelist->authorizer` for more
   details). Returns :authorized if the request is allowed, otherwise a
   string describing the reason not."
-  [whitelist]
-  {:pre  [(string? whitelist)]
+  [allowlist]
+  {:pre  [(string? allowlist)]
    :post [(fn? %)]}
-  (let [allowed? (kitchensink/cn-whitelist->authorizer whitelist)]
+  (let [allowed? (kitchensink/cn-whitelist->authorizer allowlist)]
```
Comment (austb) [inline src/puppetlabs/puppetdb/middleware.clj:52]: Do we need to do an X release of kitchensink?

File: src/puppetlabs/puppetdb/schema.clj [hunk 2]
```diff
@@ -155,6 +156,31 @@
...
+    (when (and facts-blacklist facts-blocklist)
+      (let [msg (str
+                 "Confusing configuration settings found! "
+                 "Both the deprecated facts-blacklist and "
+                 "replacement facts-blocklist are set. "
+                 "These settings are mutually exclusive, "
+                 "please prefer facts-blocklist.")]
+        (throw (ex-info msg {:type ::cli-error :message msg}))))
+    (when facts-blacklist
+      (log/warn (str "The facts-blacklist and facts-blacklist-type "
```
Comment (austb) [inline src/puppetlabs/puppetdb/schema.clj:175]: marking a string

File: src/puppetlabs/puppetdb/config.clj [hunk 3]
```diff
@@ -410,10 +422,33 @@
         warn-if-mismatched-node-purge-ttls
         forbid-duplicate-write-db-subnames)))
 
+(defn convert-certificate-whitelist-to-allowlist
+  [config]
+  (let [{:keys [certificate-whitelist
+                certificate-allowlist]} (:puppetdb config)
+        allowlist-value (or certificate-allowlist certificate-whitelist)]
+    (when (and certificate-allowlist certificate-whitelist)
+      (let [msg (str
```
Comment (austb) [inline src/puppetlabs/puppetdb/config.clj:431]: marking a string

File: src/puppetlabs/puppetdb/config.clj [hunk 4]
```diff
@@ -410,10 +422,33 @@
...
+                 "Confusing configuration settings found! "
+                 "Both the deprecated certificate-whitelist and "
+                 "replacement certificate-allowlist are set. "
+                 "These settings are mutually exclusive, "
+                 "please prefer certificate-allowlist.")]
+        (throw (ex-info msg {:type ::cli-error :message msg}))))
+    (when certificate-whitelist
+      (log/warn (str "The certificate-whitelist setting has been deprecated "
```
Comment (austb) [inline src/puppetlabs/puppetdb/config.clj:439]: marking a string

**Review by austb — CHANGES_REQUESTED** [review]
Looks good, I think we just need to i18n those 4 strings. I think I marked all of them with comments.

And one question about if we need to do an X release of kitchensink so we can get rid of that whitelist fn name

File: src/puppetlabs/puppetdb/schema.clj [hunk 5]
```diff
@@ -155,6 +156,31 @@
...
 
+(defn convert-blacklist-settings-to-blocklist [config]
+  (let [{:keys [facts-blocklist
+                facts-blocklist-type
+                facts-blacklist
+                facts-blacklist-type]} config
+        blocklist-value (or facts-blocklist facts-blacklist)
+        bloclist-type (or facts-blocklist-type facts-blacklist-type)]
+    (when (and facts-blacklist facts-blocklist)
+      (let [msg (str
```
[the remark this answers is not in the document]
Comment (Zak-Kent) [reply src/puppetlabs/puppetdb/schema.clj:167]: cool, I'll update the strs for the i18n stuff now

File: src/puppetlabs/puppetdb/middleware.clj (hunk 1 above)
Comment (Zak-Kent) [reply src/puppetlabs/puppetdb/middleware.clj:52]: I think we will need a release of kitchensink to get rid of this func name.

**Comment by Zak-Kent** [comment]
This has been updated and is ready for a 2nd look.

Once this goes in we may want to cherry-pick some of the changes back to 5.2.x and update the docs there as well. I didn't start from 5.2.x due to the changes in config handling between branches and it about to EOL.

**Review by austb — APPROVED** [review]
:+1:

**Pull request merged**